Updates to AWS AgentCore README and scripts

ronjer30 opened the pull request:
Addresses use of env variables in AWS Strands example, specifically AgentCore docker build, scripts and readme.   These have been updated to move env variables to runtime invocation

## Description
<!-- Note: The pull request title will be included in the CHANGELOG. -->
<!-- Provide a standalone description of changes in this PR. -->
<!-- Reference any issues closed by this PR with "closes #1234". All PRs should have an issue they close-->
Closes

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Updated title/branding, reordered steps, expanded prerequisites, and replaced direct-key instructions with a Secrets Manager–based credential workflow and runtime ID guidance.

* **New Features**
  * Added helper scripts and run variants (with and without OpenTelemetry) to discover, run, update, and test runtimes; credentials loaded from Secrets Manager.

* **Improvements**
  * Switched to environment-driven, dynamic container/runtime URIs, exported runtime identifiers for downstream use, Docker build optimizations, and IAM guidance for secrets access.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ronjer30 committed:
Update AWS AgentCore README and scripts to address use of env variables in docker build

Env variables are now passed to AgentCore during runtime.

Co-authored-by: Chris Smith <smithzgg@amazon.com>
Signed-off-by: RanjitR <ranjitr@nvidia.com>

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
```diff
@@ -783,23 +785,20 @@ ENTRYPOINT ["sh", "-c", "exec opentelemetry-instrument nat serve --config_file=$
 ### Credential Management
 
-**NEVER hard-code credentials in your `Dockerfile` or source code.** Always use secure credential management:
+**NEVER hardcode credentials in your `Dockerfile` or source code.** Always use secure credential management:
```
for consistency with rest of repo we use "hard-code" or else precommit checks will fail

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
```diff
@@ -783,23 +785,20 @@ ENTRYPOINT ["sh", "-c", "exec opentelemetry-instrument nat serve --config_file=$
 | ❌ Never Use | ✅ Use Instead |
 |-------------|---------------|
-| Hard-coded API keys in `Dockerfile` | AWS Secrets Manager |
+| Hardcoded API keys in `Dockerfile` | AWS Secrets Manager |
```
same as above

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
```diff
@@ -827,7 +826,7 @@ ENV AWS_ACCESS_KEY_ID="AKIAxxxxx"
 ### Action Items Before Deployment
 
-- [ ] Remove all hard-coded credentials from code
+- [ ] Remove all hardcoded credentials from code
```
same as above

ronjer30 committed:
Update AWS AgentCore README and scripts to remove env variables in AWS deployment and use Secrets Manager

Co-authored-by: Chris Smith <smithzgg@amazon.com>
Signed-off-by: RanjitR <ranjitr@nvidia.com>

ronjer30 committed:
Updated IAM policy in AgentCore readme to the right scope, add hints and changed exception type

Co-authored-by: Chris Smith <smithzgg@amazon.com>
Signed-off-by: RanjitR <ranjitr@nvidia.com>

ronjer30 committed:
refactor(strands/bedrock_agentcore): Complete deployment refactor with optimized Docker image

- Refactor all deployment scripts for improved agentcore integration
- Add new runtime scripts: run_nat_no_OTEL.sh, run_nat_with_OTEL.sh
- Add get_agentcore_runtime_id.py for runtime management
- Optimize Dockerfile: remove embedded nat serve, reduce image size
- Add .dockerignore to keep image under 4GB agentcore limit
- Update all existing scripts: deploy_nat.py, test_nat.py, update_nat.py
- AgentCore README.md updates for new deployment process

Co-authored-by: Chris Smith <smithzgg@amazon.com>
Signed-off-by: RanjitR <ranjitr@nvidia.com>

bbednarski9 committed:
README.md vale checks

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 committed:
README.md vale checks

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/scripts/update_nat.py:
```diff
@@ -12,20 +13,54 @@
+        raise Exception(f"Error retrieving secret: {e}") from e
+
+    secret = get_secret_value_response['SecretString']
+    return json.loads(secret)
```
NA

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/scripts/update_nat.py:
```diff
@@ -12,20 +12,54 @@
+        raise RuntimeError(f"Error retrieving secret: {e}") from e
+
+    secret = get_secret_value_response['SecretString']
+    return json.loads(secret)
```
NA

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
````diff
@@ -34,9 +34,66 @@ Before you begin, ensure you have the following installed:
+    export AWS_DEFAULT_REGION=$(aws configure get default.region)
+```
+### Set Account for local configurion
+Replacce <YOUR_ACCOUNT_ID HERE> with your AWS account number (example: 211123456789)
````
fixed

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
```diff
@@ -34,9 +34,66 @@ Before you begin, ensure you have the following installed:
+- AWS CLI installed and configured
+- Appropriate IAM permissions to create secrets in AWS Secrets Manager
+- Your NVIDIA API key
+
```
changed

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
````diff
@@ -573,7 +682,7 @@ After the role is created, you'll be redirected to the Roles page:
-```plaintext
+```
 arn:aws:iam::<AWS_ACCOUNT_ID>:role/AgentCore_NAT
 ```
````
resolved

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/README.md:
```diff
@@ -700,141 +809,58 @@ RUN --mount=type=cache,id=uv_cache,target=/root/.cache/uv,sharing=locked \
+# Example:
+# OTEL_RESOURCE_ATTRIBUTES=service.version=1.0,service.name=mcp-calculator,aws.log.group.names=mcp/mcp-calculator-logs
+#ENV OTEL_RESOURCE_ATTRIBUTES=service.name=nat_test_agent,aws.log.group.names=/aws/bedrock-agentcore/runtimes/<AGENTCORE_RUNTIME_ID>
+ENV OTEL_RESOURCE_ATTRIBUTES=service.name=nat_test_agent,aws.log.group.names=/aws/bedrock-agentcore/runtimes/strands_test_demo-oNUmOg6xk0
+
+# CloudWatch Integration (ensure the log group and log stream are pre-created and exists)
+# Example:
+# OTEL_EXPORTER_OTLP_LOGS_HEADERS=x-aws-log-group=mcp/mcp-calculator-logs,x-aws-log-stream=default,x-aws-metric-namespace=mcp-calculator
+#ENV OTEL_EXPORTER_OTLP_LOGS_HEADERS=x-aws-log-group=/aws/bedrock-agentcore/runtimes/<AGENTCORE_RUNTIME_ID>,x-aws-log-stream=otel-rt-logs,x-aws-metric-namespace=strands_demo
+ENV OTEL_EXPORTER_OTLP_LOGS_HEADERS=x-aws-log-group=/aws/bedrock-agentcore/runtimes/strands_test_demo-oNUmOg6xk0,x-aws-log-stream=otel-rt-logs,x-aws-metric-namespace=strands_demo
```
outdated this has been addressed

bbednarski9 committed:
coderabbit fixes

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 commented on examples/frameworks/strands_demo/bedrock_agentcore/scripts/update_nat.py:
```diff
@@ -13,19 +13,41 @@
-    agentRuntimeArtifact={
-        'containerConfiguration': {
-            'containerUri': '<AWS_ACCOUNT_ID>.dkr.ecr.<AWS_REGION>.amazonaws.com/strands-demo:latest'
-        }
-    },
-    networkConfiguration={"networkMode": "PUBLIC"},
-    roleArn='<IAM_AGENTCORE_ROLE>')
+# Configuration
+CONTAINER_IMAGE = 'strands-demo:latest'
+IAM_AGENTCORE_ROLE = '<IAM_AGENTCORE_ROLE>'
+
+AWS_REGION = os.environ['AWS_DEFAULT_REGION']
+AWS_ACCOUNT_ID = os.environ['AWS_ACCOUNT_ID']
+IAM_AGENTCORE_ROLE = f'arn:aws:iam::{os.environ.get("AWS_ACCOUNT_ID")}:role/AgentCore_NAT'
+
+RUNTIME_NAME = "strands-demo"
+#AGENT_RUNTIME_ID = os.environ['AGENT_RUNTIME_ARN']
+
+cclient = boto3.client('bedrock-agentcore-control', region_name=AWS_REGION)
+cresponse = cclient.list_agent_runtimes()
+
+for runtime in cresponse['agentRuntimes']:
+    if runtime['agentRuntimeName'] == RUNTIME_NAME:
+        runtime_id = runtime['agentRuntimeId']
+        print(f"Found runtime ID: {runtime_id}")
+        break
+
+client = boto3.client('bedrock-agentcore-control', region_name=AWS_REGION)
+
+response = client.update_agent_runtime(agentRuntimeId=runtime_id,
+                                       agentRuntimeArtifact={
+                                           'containerConfiguration': {
+                                               'containerUri': (f'{AWS_ACCOUNT_ID}.dkr.ecr.{AWS_REGION}'
+                                                                f'.amazonaws.com/{CONTAINER_IMAGE}')
+                                           }
+                                       },
+                                       networkConfiguration={"networkMode": "PUBLIC"},
+                                       roleArn=IAM_AGENTCORE_ROLE)
```
resolved

bbednarski9 committed:
NAT capitalization

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 committed:
consistent placeholders

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 committed:
Vale

Signed-off-by: Bryan Bednarski <bbednarski@nvidia.com>

bbednarski9 commented:
/ok to test 16835ef

bbednarski9 commented:
/ok to test 16835ef

bbednarski9 commented:
/ok to test 16835ef

bbednarski9 committed:
Merge branch 'develop' into feature/strands-agentcore-updates

bbednarski9 commented:
/merge
